src/main/kotlin/cacophony/codegen/instructions/CacophonyInstructionCovering.kt

```diff
@@ -2,14 +2,41 @@ package cacophony.codegen.instructions
 
 import cacophony.codegen.BlockLabel
 import cacophony.codegen.instructions.matching.InstructionMatcher
+import cacophony.codegen.instructions.matching.Match
 import cacophony.controlflow.CFGNode
+import cacophony.controlflow.Register
+
+class CacophonyInstructionCovering(private val instructionMatcher: InstructionMatcher) : InstructionCovering {
+    private fun coverGivenMatch(match: Match): List<Instruction> {
+        val tempRegisters = match.toFill.keys.associateWith { Register.VirtualRegister() }
+
+        val instructions = mutableListOf<Instruction>()
+        match.toFill.forEach { (label, node) ->
+            instructions += coverWithInstructionsForValue(node, tempRegisters[label]!!)
+        }
+        instructions += match.instructionMaker(tempRegisters)
```

Comment:
NIT: For a more functional vibe, consider replacing the `forEach` and `+=` with `match.toFill.map { ... }.flatten() + ...` :)